functions/backup.bash

```diff
@@ -325,19 +240,109 @@ create_amanda_config() {
+  successText="Setup was successful.\\n\\nAmanda backup tool is now taking backups around 01:00. For further readings, start at http://wiki.zmanda.com/index.php/User_documentation."
+
+  echo -n "$(timestamp) [openHABian] Beginning setup of the Amanda backup system... "
+  if (whiptail --title "Amanda backup installation" --yes-button "Continue" --no-button "Cancel" --defaultno --yesno "$queryText" 24 80); then echo "OK"; else echo "CANCELED"; return 0; fi
+
+  if ! dpkg -s 'exim4' &> /dev/null; then
+    if (whiptail --title "MTA missing?" --yes-button "Install EXIM4" --no-button "Ignore" --yesno "$eximText" 12 80); then
+      if ! exim_setup; then return 1; fi
+    fi
+  fi
+
+  echo -n "$(timestamp) [openHABian] Configuring Amanda backup system prerequisites... "
+  adminMail="$(whiptail --title "Amanda backup reports?" --inputbox "\\nEnter the eMail address to send backup reports to:" 9 80 3>&1 1>&2 2>&3)"
+  if [[ -z $adminMail ]]; then
+    adminMail="root@${HOSTNAME}"
+  fi
+  while [[ -z $backupPass ]]; do
+    if ! backupPass1="$(whiptail --title "Authentication Setup" --passwordbox "\\nEnter a password for ${backupUser}:" 9 80 3>&1 1>&2 2>&3)"; then echo "CANCELED"; return 0; fi
+    if ! backupPass2="$(whiptail --title "Authentication Setup" --passwordbox "\\nPlease confirm the password:" 9 80 3>&1 1>&2 2>&3)"; then echo "CANCELED"; return 0; fi
+    if [[ $backupPass1 == "$backupPass2" ]] && [[ ${#backupPass1} -ge 8 ]] && [[ ${#backupPass2} -ge 8 ]]; then
+      backupPass="$backupPass1"
+    else
+      whiptail --title "Authentication Setup" --msgbox "Password mismatched, blank, or less than 8 characters... Please try again!" 7 80
+    fi
+  done
+
+  if ! amanda_install "$backupPass"; then return 1; fi
+
+  if (whiptail --title "Backup using locally attached storage?" --yes-button "Yes" --no-button "No" --yesno "Would you like to setup a backup mechanism based on locally attached or NAS mounted storage?" 8 80); then
+    config="openhab-dir"
+    if ! storageLoc="$(whiptail --title "Storage directory?" --inputbox "\\nWhat is the directory backups should be stored in?\\n\\nYou can specify any locally accessible directory, no matter if it's located on the internal SD card, an external USB-attached device such as a USB stick, HDD, or a NFS/CIFS share." 13 80 3>&1 1>&2 2>&3)"; then echo "CANCELED"; return 0; fi
+    tapes="15"
+    if ! tapeSize="$(whiptail --title "Storage capacity?" --inputbox "\\nHow much storage do you want to dedicate to your backup in megabytes?\\n\\nRecommendation: 2-3 times the amount of data to be backed up." 11 80 3>&1 1>&2 2>&3)"; then echo "CANCELED"; return 0; fi
+    ((tapeSize/=tapes))
+    if ! create_amanda_config "$config" "$backupUser" "$adminMail" "$tapes" "$tapeSize" "$storageLoc"; then return 1; fi
+    whiptail --title "Amanda config setup successful" --msgbox "$successText" 10 80
+  fi
```

Comment:
Use `elif` to prevent multiple configs at once, which as we discussed earlier in #1055 se do not support. Unless this changes that.

Reply:
I think it does now, no?
it's separate calls to create_amanda_config, and separate timer/services to start
Did I overlook anything ?

Reply:
I will, it looks fine to me so I am ok with it. I'll let you know if I run into any issues when I test.